File: playground/TestMathFloorInfinity.cs
using Asynkron.JsEngine;

var engine = new JsEngine();

// Test Math.floor(Infinity)
var result1 = await engine.Evaluate("Math.floor(Infinity)");
Console.WriteLine($"Math.floor(Infinity) = {result1}");
Console.WriteLine($"Type: {result1?.GetType()}");
Console.WriteLine($"Is Infinity: {double.IsPositiveInfinity((double)result1!)}");

// Test comparison
var result2 = await engine.Evaluate("Math.floor(Infinity) === Infinity");
Console.WriteLine($"\nMath.floor(Infinity) === Infinity: {result2}");

var result3 = await engine.Evaluate("Math.floor(Infinity) != Infinity");
Console.WriteLine($"Math.floor(Infinity) != Infinity: {result3}");

// Test isNaN
var result4 = await engine.Evaluate("isNaN(Infinity)");
Console.WriteLine($"\nisNaN(Infinity): {result4}");

// Test the checkValue function from the test
var checkValueScript = @"
function checkValue(value){
  if(Math.floor(value)!=value||isNaN(value)){
    throw ('INVALID_INTEGER_VALUE: ' + value);
  }
  else{
    return value;
  }
}
checkValue(Infinity);
";

try
{
    var result5 = await engine.Evaluate(checkValueScript);
    Console.WriteLine($"\ncheckValue(Infinity) succeeded: {result5}");
}
catch (Exception ex)
{
    Console.WriteLine($"\ncheckValue(Infinity) failed: {ex.Message}");
}

File: playground/test-math-floor-infinity.js
// Test Math.floor(Infinity) behavior
console.log('Testing Math.floor(Infinity):');
console.log('Math.floor(Infinity) =', Math.floor(Infinity));
console.log('Infinity =', Infinity);
console.log('Math.floor(Infinity) === Infinity:', Math.floor(Infinity) === Infinity);
console.log('Math.floor(Infinity) != Infinity:', Math.floor(Infinity) != Infinity);
console.log('isNaN(Infinity):', isNaN(Infinity));

// This is the check from the test
function checkValue(value){
  console.log('\ncheckValue called with:', value);
  console.log('Math.floor(value):', Math.floor(value));
  console.log('value:', value);
  console.log('Math.floor(value) != value:', Math.floor(value) != value);
  console.log('isNaN(value):', isNaN(value));

  if(Math.floor(value) != value || isNaN(value)){
    throw ("INVALID_INTEGER_VALUE: " + value);
  }
  else{
    return value;
  }
}

try {
  var result = checkValue(Infinity);
  console.log('SUCCESS: checkValue(Infinity) returned', result);
} catch(e) {
  console.log('ERROR:', e);
}
